src/Workspaces/Core/Portable/Workspace/Solution/TextLoader.cs

```diff
@@ -23,6 +23,14 @@ public abstract class TextLoader
 
         internal virtual string? FilePath => null;
 
+        internal virtual SourceHashAlgorithm ChecksumAlgorithm => SourceHashAlgorithm.Sha1;
+
+        internal TextLoader? TryUpdateChecksumAlgorithm(SourceHashAlgorithm algorithm)
+            => (ChecksumAlgorithm == algorithm) ? this : TryUpdateChecksumAlgorithmImpl(algorithm);
+
+        private protected virtual TextLoader? TryUpdateChecksumAlgorithmImpl(SourceHashAlgorithm algorithm)
+            => null;
```

Comment:
So this seems pretty icky that the text loaders have state like this that we have to update, and the fact this is private protected makes me wonder what other implementers are supposed to do. Given you're _already_ making API changes to TextLoader in your other PR where we're removing the Workspace attributes, what if combine the efforts and just pass the checksum algorithm to the LoadTextAndVersion call? As a future-proofing I'd say we make a little struct that we pass instead so if we need to pass something _else_ down the road we can.

This change is obviously out of scope for this PR (if nothing else, we'd need the other PR merged), but thoughts? This also avoids the ickiness of DocumentInfo having to read checksum algorithms from TextLoader, etc.

Reply:
Yes, definitely something we can try as a follow up. I agree it would simplify things.